### DISCUSSION TITLE
Any ztest/twister automated test cases available to test Ethernet Driver

### DISCUSSION THREAD

**Discussion by deepak-shreshti** [zephyrproject-rtos/zephyr#55580] [Q&A] [answered]
Dear Zephyr-team,  

We are currently porting Zephyr-OS to a specific MCU Board. 
Base development is completed. 
We could add SOC and board specific files and basic test is PASS.
We added Ethernet driver support and have tested ethernet driver using UDP/TCP sample application. We would like to know ztest/twister automated test cases available for Ethernet Driver?
Basically we would like to know mandatory ztest/twister cases for ethernet 
that should be passed to register or Qualify Zephyr-os? 

Regards
Deepak

**Comment by jukkar** [comment, marked as answer]
All the network tests in `tests/net` use simulated network drivers for testing, so they are not really useful to you as your network driver would not be used by those tests even if you run those tests in your board.
You would need to create tests for your Ethernet driver in `tests/drivers` directory. Not sure what kind of tests those would be, perhaps sending and receiving data and verifying that data is correct or not. I suppose it is your to decide how / what you want to test in your driver.

**Reply by deepak-shreshti** [reply]
Dear jukkar,

Thank you for the response. We understand that we have to create test cases to test ethernet send/receive.
We would like to know if there are any ztest cases available for other peripherals such as uart, spi, i2c, adc etc.

Regards
Deepak

**Comment by jukkar** [comment]
Please check `tests/drivers` directory, it contains many tests for various of drivers in the system. If your board is missing from the test, then just add one in the corresponding `boards/` directory.
